data/scripts.js

```diff
@@ -88,6 +88,28 @@ exports.BattleScripts = {
 		this.useMove(baseMove, pokemon, target, sourceEffect, zMove);
 		this.singleEvent('AfterMove', move, null, pokemon, target, move);
 		this.runEvent('AfterMove', pokemon, target, move);
+
+		// Dancer's activation order is completely different from any other event, so it's handled separately
+		if (move.flags['dance']) {
+			let dancers = [];
+			for (const side of this.sides) {
+				for (const slot of side.active) {
+					let currentPoke = slot;
+					if (!currentPoke || !currentPoke.hp || pokemon === currentPoke || move.isExternal) continue;
+					if (currentPoke.ability === 'dancer') {
```

Comment:
This needs to be `if (currentPoke.hasAbility('dancer')) {` in case of Gastro Acid/Core Enforcer.